loginterceptor/loginterceptor.go

```diff
@@ -0,0 +1,92 @@
+package loginterceptor
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"sync"
+)
+
+// PrefixInterceptor intercept writes: if a line begins with prefix, it will be written to
+// both writers. Partial writes without newline are buffered until a newline.
+type PrefixInterceptor struct {
+	re          *regexp.Regexp
+	intercepted io.Writer
+	original    io.Writer
+
+	// internal pipe and goroutine to scan and route
+	pr *io.PipeReader
```

Comment:
Please don't use short and very similar (tr, pr, ti, tw, ... ) variable names in fields, that makes understanding the code really hard